Update Policy documentation to support UAMI and new remediation properties (#84699)
Azure/azure-content: articles/governance/policy/how-to/remediate-resources.md

## How remediation security works

> When Azure Policy runs the template in a **deployIfNotExists** policy definition or runs **modify operations**, it does so using

Reviewer: ```suggestion
When Azure Policy starts a template deployment when evaluating **deployIfNotExists** policies or modifies a resource  when evaluating **modify** policies, it does so using
```

> a [managed identity](../../../active-directory/managed-identities-azure-resources/overview.md).

Reviewer: ```suggestion
a [managed identity](../../../active-directory/managed-identities-azure-resources/overview.md) that is associated with the policy assignment.
```

> Azure Policy allows the option to either create (system assigned) or select (user assigned) a managed identity  for each assignment, but the identity must have details about what roles

Reviewer: ```suggestion
Policy assignments can either use a system assigned managed identity that is created by the policy service or a user assigned identity provided by the user.  The managed identity needs to be granted the appropriate roles required for remediating resources
```

Reviewer: Or something along that lines... "but the identity must have details about what roles to grant the managed identity" is just weird IMO.

to grant the managed identity. If the managed identity is missing roles, an error is displayed
during the assignment of the policy or an initiative. When using the portal, Azure Policy
automatically grants the managed identity the listed roles once assignment starts. When using SDK,
the roles must manually be granted to the managed identity. The _location_ of the managed identity
doesn't impact its operation with Azure Policy.

> :::image type="content" source="../media/remediate-resources/remediation-tab.png" alt-text="Screenshot of a policy assignment creating a system assigned managed identity in East US wit Log Analytics Contributor permissions." border="false":::

Reviewer: `with`?

Author: Changed.

## Manually configure the managed identity

When creating an assignment using the portal, Azure Policy can both generate a managed identity and

> grants it the roles defined in **roleDefinitionIds**. In the following conditions, steps to create

Reviewer 2: ```suggestion
grant it the roles defined in **roleDefinitionIds**. In the following conditions, steps to create
```

the managed identity and assign it permissions must be done manually:

- While using the SDK (such as Azure PowerShell)
- When a resource outside the assignment scope is modified by the template
- When a resource outside the assignment scope is read by the template

## Configure a managed identity through Portal

When creating an assignment using the portal, it is possible to select either a system or user assigned managed identity. 

To set a system assigned managed identity through portal follow these steps: 

1. In the **Remediation** tab of the create/edit assignment view, ensure that **System assigned managed identity** 
is selected under "Types of Managed Identity". 

> 1. Specify the location at which the managed identity is to be located. 

Reviewer 2: ```suggestion
2. Specify the location at which the managed identity is to be located. 
```

Author: It seems that the docs will auto change the numbers. I believe this a doc pattern to just name everything one.

To set a user assigned managed identity through portal follow these steps: 

1. In the **Remediation** tab of the create/edit assignment view, ensure that **User assigned managed identity** 
is selected under "Types of Managed Identity". 

> 1. Specify the scope where the managed identity is hosted. The scope of the managed identity does not have to equate to the scope of the assignment, but has to be in the same tenant. 

Reviewer 2: ```suggestion
2. Specify the scope where the managed identity is hosted. The scope of the managed identity does not have to equate to the scope of the assignment, but has to be in the same tenant. 
```

> 1. Select the managed identity under "Existing user assigned identities". 

Reviewer 2: ```suggestion
3. Select the managed identity under "Existing user assigned identities". 
```

 > [!NOTE]

>    > If the managed identity does not have the permissions need to execute the required remediation task,

Reviewer 2: ```suggestion
   > If the managed identity does not have the permissions needed to execute the required remediation task,
```

   > it will be granted permissions *automatically* only through the portal. 

>    > For all methods, permissions must be configure manually. 

Reviewer 2: ```suggestion
   > For all other methods, permissions must be configured manually. 
```

### Create managed identity with PowerShell

> To create a identity during the assignment of the policy, **Location** must be defined and

Reviewer 2: ```suggestion
To create an identity during the assignment of the policy, **Location** must be defined and
```

**Identity** used. The following example gets the definition of the built-in policy **Deploy
SQL DB transparent data encryption**, sets the target resource group, and then creates the
assignment using a **system assigned** managed identity.

```azurepowershell-interactive
# Login first with Connect-AzAccount if not using Cloud Shell

# Get the built-in "Deploy SQL DB transparent data encryption" policy definition
$policyDef = Get-AzPolicyDefinition -Id '/providers/Microsoft.Authorization/policyDefinitions/86a912f6-9a06-4e26-b447-11b16ba8659f'

# Get the reference to the resource group
$resourceGroup = Get-AzResourceGroup -Name 'MyResourceGroup'

# Create the assignment using the -Location and -Identity properties
$assignment = New-AzPolicyAssignment -Name 'sqlDbTDE' -DisplayName 'Deploy SQL DB transparent data encryption' -Scope $resourceGroup.ResourceId -PolicyDefinition $policyDef -Location 'westus' -IdentityType "SystemAssigned"
```

The following example gets the definition of the built-in policy **Deploy
SQL DB transparent data encryption**, sets the target resource group, and then creates the

> assignment using an **user assigned** managed identity.

Reviewer 2: ```suggestion
assignment using a **user assigned** managed identity.
```

Author: "An" should be used here.

# Get the reference to the resource group
$resourceGroup = Get-AzResourceGroup -Name 'MyResourceGroup'

> # Get the existing user assigned managed identity ID

Reviewer 2: This command creates the user assigned identity. We could either modify the comment to show that an identity is being created, or we can modify the powershell command to show a user assigned identity being fetched

Author: Updated

$userassignedidentity = New-AzUserAssignedIdentity -ResourceGroupName $rgname -Name $userassignedidentityname -Location $location
$userassignedidentityid = $userassignedidentity.Id

# Create the assignment using the -Location and -Identity properties
$assignment = New-AzPolicyAssignment -Name 'sqlDbTDE' -DisplayName 'Deploy SQL DB transparent data encryption' -Scope $resourceGroup.ResourceId -PolicyDefinition $policyDef -Location 'westus' -IdentityType "UserAssigned" -IdentityId $userassignedidentityid
```

The `$assignment` variable now contains the principal ID of the managed identity along with the
standard values returned when creating a policy assignment. It can be accessed through

> `$assignment.Identity.PrincipalId`.

Reviewer 2: $assignment.Identity.PrincipalId will give the principal ID for a system assigned id. For a user assigned id, we would access the principal id through $assignment.Identity.UserAssignedIdentities[$userassignedidentityid].PrincipalId

Author: addressed.

   > the **Compliance** page, then select the **Create Remediation Task** button.

1. On the **New remediation task** page, optional remediation settings are shown: 

1. **Failure Threshold percentage** used to specifying whether the remediation task should fail if the percentage of failures exceeds the given threshold. Provided as a number between 0 to 100. By default, the failure threshold is 100%. 
2. **Resource Count** determines how many non-compliant resources to remediate in a given remediation task. The default value is 500 (the previous limit). The maxium number of is 10,000 resources.
3. **Parallel Deployments** determines how many resources to remediate at the same time. The allowed values are 1 to 15 resources at a time. The default value is 10. 

   > [!NOTE]
   > These settings cannot be changed once the remediation task has started.
 
1.  On the same page, filter the resources to remediate by using the **Scope**
   ellipses to pick child resources from where the policy is assigned (including down to the
   individual resource objects). Additionally, use the **Locations** dropdown list to further filter

>    the resources. The table will only show a subset (top 250) resources. 

Reviewer: Maybe instead of this sentence, just add a note saying that if the number of non-complaint resources that matches the selected filter is too large, the table will only show the _first_ (I don't like the word top, as it implies some sort of order) 250 non-compliant resources.
Alternatively just drop this part altogether. We're already show an info box in the UX in this case, there's no need to cover it here. It's too much information IMO.

Author: Dropped.

   :::image type="content" source="../media/remediate-resources/task-progress.png" alt-text="Screenshot of the Remediation tasks tab and progress of existing remediation tasks." border="false":::

1. Select on the **remediation task** from the policy compliance page to get details about the

>    progress. The filtering used for the task is shown along with status and a list of the top 500 resources being

Reviewer: Same as above, I don't think it's worth mentioning since this is an edge case for large remediation and when it happens, we show an info box to the user. No need to document all of the UX edge cases if the UX is already explaining them when they happen :)

Author: Dropped the top 500.
